models/BaseEntity.cfc

```diff
@@ -3008,23 +3148,45 @@ component accessors="true" {
 					);
 
 					param meta.originalMetadata.properties = [];
-					meta[ "attributes" ]                   = generateAttributesFromProperties( meta.originalMetadata.properties );
-					for ( var key in keyNames() ) {
+
+					meta[ "attributes" ] = generateAttributesFromProperties(
+						meta.hasParentEntity ? meta.localMetadata.properties : meta.originalMetadata.properties
+					);
+					if ( structKeyExists( meta.localMetadata, "discriminatorColumn" ) ) {
+						meta.attributes[ meta.localMetaData.discriminatorColumn ] = paramAttribute( { "name" : meta.localMetaData.discriminatorColumn } );
+					}
+					arrayWrap( variables._key ).each( function( key ) {
 						if ( !meta.attributes.keyExists( key ) ) {
 							var keyProp                     = paramAttribute( { "name" : key } );
 							meta.attributes[ keyProp.name ] = keyProp;
 						}
-					}
+					} );
 					meta[ "casts" ] = generateCastsFromProperties( meta.originalMetadata.properties );
 					guardKeyHasNoDefaultValue( meta.attributes );
 					return meta;
 				} )
 			);
 		}
 
-		variables._fullName           = variables._meta.fullName;
-		variables._entityName         = variables._meta.entityName;
-		variables._table              = variables._meta.table;
+		variables._fullName        = variables._meta.fullName;
+		variables._entityName      = variables._meta.entityName;
+		variables._table           = variables._meta.table;
+		variables._hasParentEntity = variables._meta.hasParentEntity;
+
+		if ( variables._hasParentEntity ) {
+			variables._parentEntity = variables._meta.parentEntity;
+
+			if ( structKeyExists( variables._parentEntity, "discriminatorColumn" ) ) {
+				variables[ "get" & variables._parentEntity.discriminatorColumn ] = function() {
```

Comment:
Instead of adding it here as a function, let's add a check to `retrieveAttribute` or in our `tryAttributesGetter` call.  This way a user can still override the function if they would like.  If we need to call the getter ourselves internally, we should probably store this value in some private state (prefixed with an `_`).

Reply:
Good call.  I'll take a look at how that's structured and move it over.

Reply:
Actually, this closure creation is not needed as the discriminator column is already scoped as a attribute and the `onMissingMethod` fall through handles it just fine. :)   Removed.